include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1935,8 +1945,11 @@ __parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPoli
         // We shouldn't have any restrictions for _AtomicType type here
         // because we have a single work-group and we don't need to use atomics for inter-work-group communication.
 
+        using _KernelName =
```

Comment:
You use the same name in both if and else branches. Today it works but in principle, it's not necessary. I would suggest to rename it to the proper name for easier refactoring in the future.

```suggestion
        using __find_or_one_wg_kernel_name = 
```